pkg/pillar/cmd/zedagent/parsepatchenvelopes.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2023 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package zedagent
+
+import (
+	"encoding/hex"
+	"fmt"
+	"os"
+
+	"crypto/sha256"
+
+	zconfig "github.com/lf-edge/eve-api/go/config"
+	"github.com/lf-edge/eve/pkg/pillar/persistcache"
+	"github.com/lf-edge/eve/pkg/pillar/types"
+
+	"github.com/lf-edge/eve/pkg/pillar/utils/generics"
+)
+
+func parsePatchEnvelopes(ctx *getconfigContext, config *zconfig.EdgeDevConfig) {
+	parsePatchEnvelopesImpl(ctx, config, types.PersistCachePatchEnvelopes)
+}
+
+func parsePatchEnvelopesImpl(ctx *getconfigContext, config *zconfig.EdgeDevConfig,
+	persistCacheFilepath string) {
+	log.Tracef("Parsing patchEnvelope from configuration")
+
+	// Remove previously created patch envelopes
+	// so that we will not have stale objects
+	if err := os.RemoveAll(persistCacheFilepath); err != nil {
+		log.Errorf("Failed to delete persistCacheFilepath %v", err)
+		return
+	}
+
+	// Store list of binary blobs which were created before
+	pc, err := persistcache.New(persistCacheFilepath)
+	if err != nil {
+		log.Errorf("Failed to load persistCache %v", err)
+		return
+	}
+	blobsBefore := pc.Objects()
+
+	var blobsAfter []string
+	patchEnvelopes := config.GetPatchEnvelopes()
+	result := types.PatchEnvelopes{}
+	for _, pe := range patchEnvelopes {
```

Comment:
Do we destroy and re-create patch envelopes every time configuration gets pulled from zedcloud? Does it happen even when the patch envelope did not change?

Reply:
We destroy only inline objects, which are not referenced anymore when new configuration comes. If inline patch envelope did not change we will not override it.